sigopt_sklearn/search.py

```diff
@@ -150,6 +150,10 @@ def __init__(self, estimator, param_domains, n_iter=10, scoring=None,
         self.opt_timeout = opt_timeout
         self.verbose = verbose
 
+        # Stores the mappings between categorical strings to Python values. The keys correspond to parameter names and
+        # values correspond to the string-to-value mappings themselves.
+        self.categorical_mappings_ = {}
```

Comment:
instead of special-casing categoricals, maybe we should just construct this mapping for all parameter types. and then for ints/doubles it's just an identity function, and for categoricals is a .get from a map, where that map is what the user provides, or {X: X} if they specify their categoricals as a list